axiom/datasets.go

```diff
@@ -331,6 +333,60 @@ func (s *DatasetsService) IngestEvents(ctx context.Context, id string, opts Inge
 	return &res, nil
 }
 
+// IngestChannel ingests events from a channel into the dataset identified by
+// its id. As it keeps a connection open until the channel is closed, it is not
+// advised to use this method for long-running ingestions.
+//
+// Restrictions for field names (JSON object keys) can be reviewed here:
+// https://www.axiom.co/docs/usage/field-restrictions.
+func (s *DatasetsService) IngestChannel(ctx context.Context, id string, events <-chan Event, opts IngestOptions) (*IngestStatus, error) {
+	path, err := addOptions(s.basePath+"/"+id+"/ingest", opts)
+	if err != nil {
+		return nil, err
+	}
+
+	pr, pw := io.Pipe()
+	go func() {
+		zsw, wErr := zstd.NewWriter(pw)
+		if wErr != nil {
+			_ = pw.CloseWithError(wErr)
+			return
+		}
+
+		var (
+			enc    = json.NewEncoder(zsw)
+			encErr error
+		)
+		for event := range events {
+			if encErr = enc.Encode(event); encErr != nil {
+				break
+			}
```

Comment:
Do we need a check for ctx.Done() here? Reading from the events pipe can take however long and our ctx might have been cancelled...